errors/public.go

```diff
@@ -60,3 +62,14 @@ func GetName(err error) string {
 		return "ErrUnknown"
 	}
 }
+
+// ElementError adds generic "element" information (file path, workflow job, step ID,
+// etc) to an error.
+type ElementError struct {
+	Err     error
```

Comment:
if your custom error type wraps another error, you need to provide a way to unwrap it so that `errors.Is` and `errors.As` work with your type.

For example, here's a [similar wrapper in the webapp](https://github.com/ossf/scorecard-webapp/blob/623b0d90aab250ee566518dbd26238f1f1b3dfac/app/server/verify_workflow.go#L65-L75)